southworkscom/botbuilder-dotnet
# [MANX-407] Add support for ChannelData field

### Description
- Conversation references will include a channelData field.
- When calling createConversation, the channelData field will be populated with the tenant information if specified in the reference

### Changes made
- `CreateConversationAsync` now has an overload that takes a `ConversationReference` as a parameter

- `ConversationReference` now has ChannelData as a field, with the type `object`

- If the value is given `ChannelData` gets populated by an object with a TenantId `{tenant: tenantId }`


This PR corresponds to [Issue #1274](https://github.com/Microsoft/botbuilder-dotnet/issues/1274)
### To test it
* Connect a bot to Teams. Follow the steps on [this documentation](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-test)

* Using a `ConversationReference` extracted from an incoming _group_ message, use `adapter.createConversationAsync` to create a new _private_ context.

* Use the _new context_ to send a private message to the user.  Note that it is received as a **DM**, instead of a **reply** in the group channel.
